### PR TITLE
Add more tests for end twice

### PR DESCRIPTION
**Pull request by Richard-Yunchao**
The existing end twice tests covers pass end twice when its parent encoder is open and the pass encoder is valid.

This change adds more tests to cover situations below:
  - the parent encoder is locked (pass end twice in the middle of another pass)
  - the pass encoder is invalid and we end the pass twice

It should generate validation error immediately when pass end twice in the situations above.

Issue: #

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by Richard-Yunchao** [comment]
PTAL. Thanks!

File: src/webgpu/api/validation/encoding/encoder_state.spec.ts [hunk 1]
```diff
@@ -177,15 +177,20 @@ g.test('pass_end_none')
...
+g.test('pass_end_twice,basic')
+  .desc(
+    'Test that ending a {compute,render} pass twice generates a validation error. The parent encoder (command encoder) can be either locked or open.'
+  )
   .paramsSubcasesOnly(u =>
     u //
       .combine('passType', ['compute', 'render'])
+      // Simply end twice, the parent encoder is open at that time. If the second pass end is in the middle of another pass, the parent encoder is locked. It should generate a validation error in either situation.
       .combine('endTwice', [false, true])
+      .combine('secondEndInAnotherPass', [false, true])
```
Comment (kainino0x) [inline src/webgpu/api/validation/encoding/encoder_state.spec.ts:189]: nit: Just for completeness can we test both compute and render for this extra pass? (`false, 'compute', 'render'`)

**Review by kainino0x — APPROVED** [review]
LGTM! w/ one additional suggestion

File: src/webgpu/api/validation/encoding/encoder_state.spec.ts (hunk 1 above)
Comment (Richard-Yunchao) [reply src/webgpu/api/validation/encoding/encoder_state.spec.ts]: Good point. Done!

**Comment by Richard-Yunchao** [comment]
Merging now... It works fine in latest Chromium with my Dawn patch rolled in.

**Pull request merged**